### ISSUE TITLE
Project status in Netbeans 11

### ISSUE THREAD

**Issue by hamishgraham** [kelemen/netbeans-gradle-project#424]
Hello,

I see that Netbeans 11 due to be released shortly has built-in Gradle support. What is the status of this project in light of that? Will development continue or perhaps efforts re-focused on improving the built-in Gradle support?

Thanks.

**Comment by kelemen** [comment]
I assume that eventually the built-in support will replace this plugin. Though personally, I miss features (that are very-very hard to add to that) that I regularly use in this. So, I intend to provide some level of support for this plugin for the time being.

**Comment by tcfurrer** [comment]
I agree regarding the continued use of plugin features.  My (large, complex, internal-to-a-big-company) project invested a fair amount of time to get everything working well with this plugin, and we now intend to continue with that configuration until the built-in Gradle support is a little more mature and capable.  But we also want to move to Netbeans 11 because of the Java 12 support, etc.

So we appreciate the efforts to keep this plugin alive at least a little bit longer, if not more than a little bit.

**Comment by tcfurrer** [comment]
Unfortunately with my testing so far on Netbeans 11 vc4, there are several little things which stopped working, at least when running the plugin version 2.0.2.  I'm hoping these will be fixable in a new release targetted at Netbeans 11... we'll see.  I'll be filing separate new tickets for each issue I've found.

**Comment by hamishgraham** [comment]
> I assume that eventually the built-in support will replace this plugin. Though personally, I miss features (that are very-very hard to add to that) that I regularly use in this. So, I intend to provide some level of support for this plugin for the time being.

Interesting, thanks for the reply. What features are you referring to and why might they be hard to add to the built-in Gradle?

(also, thank you for all your work providing this fantastic plugin. It's meant I could keep on using Netbeans.)

**Comment by kelemen** [comment]
@tcfurrer Did you uninstall all the built-in Gradle support plugins (probably everything with Gradle in its name)? Because these two plugins are likely to conflict in various ways.

@hamishgraham Mainly the project/global properties support (which is one of the biggest part of the plugin, aside from maybe Gradle model loading). Given that subprojects are shown as separate projects, it is rather tricky in NB (a lot more tricky to do good than one would naively imagine, and would be hard to integrate with the other plugin), also the property inheritence and the configurability of tasks (say I have profiles set up where the debug tasks ssh the release package, installs it, and starts remote debugging the process). So, I'm thinking in a library which handles such properties in a generic manner, so that any GUI application could use it (so, then maybe it could be used in the new plugin).

**Comment by tcfurrer** [comment]
Yes, after uninstalling the new "Gradle" and "Groovy and Gradle" plugins, so far everything is working great for me now on Netbeans 11 with this plugin version 2.0.2.  Hopefully we can think of something in #429 to prevent others from making the same mistake I did, and getting confused.

**Comment by slominskir** [comment]
How DO you uninstall the built-in gradle plugin?  I am able to deactivate both "Gradle" and "Groovy and Gradle", but the "Uninstall" button is still disabled (grayed out) even after the IDE restarts.  Also, it might make sense to create a full list of the differences between new built-in plugin and this one.  As far as I can tell the new plugin is a strict subset (does nothing new, and a lot less).  In particular, it seems new plugin does not support setting Gradle action for "Compile on Save".

**Comment by kelemen** [comment]
I don't know all features of the built-in plugin as it is completely unrelated to this plugin.

As for uninstalling, I don't know (I can select uninstall). Maybe because the user you started NB with has no right to delete files from the installation directory?

**Comment by slominskir** [comment]
Looks like if you create a project using the new built-in plugin then you are stuck.  I can't find a way to update the project to use the old plugin.   Deleting all the netbeans related metadata in the project doesn't do it.  Maybe there is a Netbeans cache somewhere that needs to cleared.  Closing the project and reopening only after relaunching Netbeans doesn't fix it.  

Also, new plugin will gladly assume ownership of projects that were managed by old plugin if you open it before installing old, and then no way to go back.

**Comment by slominskir** [comment]
Yup, it WAS file permissions.  I right clicked and ran as administrator and then was able to Uninstall.   Thanks.

**Comment by kelemen** [comment]
The files NB (or any plugin) creates does not affect what plugin owns the project. It is detected at runtime, and this plugin does not care if other plugins generated files somewhere (and I suppose it is true for most plugins).
